src/build-from-oxlint-config.spec.ts

```diff
@@ -0,0 +1,342 @@
+import { describe, expect, it } from 'vitest';
+import {
+  buildFromOxlintConfig,
+  buildFromOxlintConfigFile,
+} from './build-from-oxlint-config.js';
+import fs from 'node:fs';
+import { execSync } from 'node:child_process';
+import type { Linter } from 'eslint';
+import { typescriptRulesExtendEslintRules } from '../scripts/constants.js';
+
+describe('buildFromOxlintConfig', () => {
+  describe('rule values', () => {
+    it('detect active rules inside "rules" scope', () => {
+      ['error', ['error'], 'warn', ['warn'], 1, [1], 2, [2]].forEach(
+        (ruleSetting) => {
+          const rules = buildFromOxlintConfig({
+            rules: {
+              eqeqeq: ruleSetting,
+            },
+          });
+
+          expect(rules.length).toBe(1);
+          expect(rules[0].rules).not.toBeUndefined();
+          expect('eqeqeq' in rules[0].rules!).toBe(true);
+          expect(rules[0].rules!.eqeqeq).toBe('off');
+        }
+      );
+    });
+
+    it('skip deactive rules inside "rules" scope', () => {
```

Comment:
After sleeping 2 days. I do not know if I want to keep this behavior. 
When someone wants to disable it in oxlint, there is 2 reasons:

- they want it disabled in oxlint and eslint
- oxlint is not aligned with eslint

Reply:
I don't have full context. Remove it if you don't feel it is correct.

Reply:
When the config is:

```
{
  "rules": {
    "eqeqeq": "off",
  }
}
```

We do not disable the rule in eslint.

Reply:
🤔 

> We do not disable the rule in eslint.

This should be the correct behaviour?

Reply:
That was I thought too.  But then I asked [why a user is disabling a rule](https://github.com/oxc-project/eslint-plugin-oxlint/pull/197#discussion_r1826442758).  
When the user do want the rule deactived in oxlint, why should it be actived in eslint?

Maybe this is only a small edge case and we can ignore it for the moment. ^^